modin/experimental/pandas/io.py

```diff
@@ -137,9 +140,13 @@ def read_custom_text(
     -------
     modin.DataFrame
     """
+    _, _, _, kwargs = inspect.getargvalues(inspect.currentframe())
+
     Engine.subscribe(_update_engine)
+    from modin.core.execution.dispatching.factories.dispatcher import FactoryDispatcher
+
     assert IsExperimental.get(), "This only works in experimental mode"
-    _, _, _, kwargs = inspect.getargvalues(inspect.currentframe())
+
```

Comment:
Is there a reason why the kwargs line was moved up?

Reply:
Yes, kwargs line was moved up to avoid passing `FactoryDispatcher` argument into kwargs. If `FactoryDispatcher` argument is present in the kwargs, it can break further io logic.